kitsune/sumo/static/sumo/js/form-wizard.js

```diff
@@ -69,9 +75,84 @@ export class FormWizard extends HTMLElement {
     this.#updateFormProgress();
   }
 
-  #updateSteps() {
-    this.#steps = [...this.children].map((child) => child.getAttribute("name"));
-    this.#updateFormProgress();
+  /**
+   * @typedef {object} FormWizardStep
+   * @property {string} name
+   *  The name of the step. Used as an identifier to determine which
+   *  child view should be shown and help determine how far the user
+   *  has progressed through the form.
+   * @property {string} status
+   *  Status of the step. One of "done" | "active" | "unavailable".
+   *  Used to hide/show steps and update the step indicator.
+   * @property {string} label
+   *  Label for the form step to be shown in the step indicator.
+   */
+
+  /**
+   * Stores the possible form step names and associated data.
+   *
+   * @param {FormWizardStep[]} nextSteps
+   *  Step data representing the desired state of the wizard.
+   */
+  set steps(nextSteps) {
+    this.#steps = nextSteps;
+    // Determine whether or not we need to update activeStep.
+    let nextActiveStep = nextSteps.find(
+      (step) => step.status === "active"
+    )?.name;
+    if (nextActiveStep) {
+      this.activeStep = nextActiveStep;
+    }
+    // Build or update the steps sidebar.
+    this.#updateStepIndicator();
+  }
+
+  /**
+   * Changes the active step and updates the status of the
+   * previously active step to "done".
+   *
+   * @param {string} name - Name of the active step.
+   * @param {FormWizardStep} data - Data about the active step.
+   */
+  setStep(name, data) {
+    let nextSteps = this.#steps.map((step) => {
+      if (step.status === "active") {
+        return { ...step, status: "done" };
+      }
+      return step.name === name ? { ...step, ...data } : step;
+    });
```

Comment:
I just realized we're not setting the active step here based on `name`. Is that something we planned for a follow-up?

Reply:
Do you mean doing something like this?

```js
{ ...step, ...data, status: "active" }
```

I could definitely do that here/as part of this patch. I wasn't totally sure if the status was going to be part of the `data` passed to `setStep`.

Reply:
Ohhh, I see. I think it'd be simpler on the manager end of things if `setStep` also did the work of making the step associated with `name` active.